external-tracer/src/lib.rs

```diff
@@ -71,6 +71,21 @@ pub fn trace(config: &TraceConfig) -> Result<Vec<GethExecTrace>, Error> {
         },
     )?;
 
-    let trace = serde_json::from_str(&trace_string).map_err(Error::SerdeError)?;
+    let trace: Vec<GethExecTrace> =
+        serde_json::from_str(&trace_string).map_err(Error::SerdeError)?;
+    // Don't throw only for specific invalid transactions we support.
+    for trace in trace.iter() {
+        if trace.invalid {
+            if !(trace.return_value.starts_with("nonce too low")
+                || trace.return_value.starts_with("nonce too high")
+                || trace.return_value.starts_with("intrinsic gas too low")
+                || trace
+                    .return_value
+                    .starts_with("insufficient funds for gas * price + value"))
+            {
+                return Err(Error::TracingError(trace.return_value.clone()));
```

Comment:
Is there an error code for this errors?

Reply:
I only see a way to get the error as a string, it also seems like in geth errors are strings as well.